operator/checkcontroller/executor.go

```diff
@@ -93,3 +108,77 @@ func (c *CheckExecutor) setupEnvVars(ctx context.Context) []corev1.EnvVar {
 func (c *CheckExecutor) cleanupOldChecks(ctx context.Context, check *k8upv1.Check) {
 	c.CleanupOldResources(ctx, &k8upv1.CheckList{}, check.Namespace, check)
 }
+
+func (c *CheckExecutor) appendTLSOptionsArgs() []string {
+	var args []string
+	if !(c.check.Spec.Backend != nil && c.check.Spec.Backend.TLSOptions != nil) {
+		return args
+	}
+
+	if c.check.Spec.Backend.TLSOptions.CACert != "" {
+		args = append(args, []string{"-caCert", c.check.Spec.Backend.TLSOptions.CACert}...)
+	}
+	if c.check.Spec.Backend.TLSOptions.ClientCert != "" && c.check.Spec.Backend.TLSOptions.ClientKey != "" {
+		addMoreArgs := []string{
+			"-clientCert",
+			c.check.Spec.Backend.TLSOptions.ClientCert,
+			"-clientKey",
+			c.check.Spec.Backend.TLSOptions.ClientKey,
+		}
+		args = append(args, addMoreArgs...)
+	}
+
+	return args
+}
+
+func (c *CheckExecutor) attachMoreVolumes() []corev1.Volume {
```

Comment:
Same here.

Reply:
I removed duplicate code. I put all of them in a utils file and wrote tests for them